### DISCUSSION TITLE
Microsoft Entra Customer Tenant

### DISCUSSION THREAD

**Discussion by rjnls** [nextauthjs/next-auth#8872] [Help]
### Summary

The new Entra offering looks a lot like B2C but in trying to use the B2C provider I get the "expected 200 and got 404" error (I wish this error would mention the url it's trying to access). My question is, should/could I hope for the B2C provider to work with the new customer tenant type from Entra? 

### Additional information

```js
Here are the node instructions: 

https://learn.microsoft.com/en-us/azure/active-directory/external-identities/customers/tutorial-web-app-node-sign-in-sign-out

It's got some things that seem new/different:

 authority: process.env.AUTHORITY || `https://${TENANT_SUBDOMAIN}.ciamlogin.com/`, // replace "Enter_the_Tenant_Subdomain_Here" with your tenant name
```

**Comment by rjnls** [comment]
No matter how I try to set up the new Entra CIAM stuff I get this error:

![image](https://github.com/nextauthjs/next-auth/assets/92457701/c14ca1a9-0645-41d9-b259-096bfd3e3bfd)

There's exactly 2 google results for that error an one seems to say this is a bug as of yesterday.

**Comment by andrem0** [comment]
Having the same issue.

**Reply by rjnls** [reply]
i could not make the ciam endpoint work. I got it working with these endpoints:
    type: 'oauth',
    wellKnown: `https://login.microsoftonline.com/${process.env.ENTRA_TENANT_ID}`,
    token: `https://login.microsoftonline.com/${process.env.ENTRA_TENANT_ID}/oauth2/v2.0/token`,

**Reply by eduardogoncalves** [reply]
HI @rjnls, 
I'm using like this and I'm able to login
![image](https://github.com/nextauthjs/next-auth/assets/4308648/10fbf63e-bd11-4e86-a1fe-c4ef14cdfa67)

but only if I use an external account like @hotmail

If I create a new account
![image](https://github.com/nextauthjs/next-auth/assets/4308648/05f27a68-7a2e-477a-bda5-eb387cc0df11)

it fails to login
![image](https://github.com/nextauthjs/next-auth/assets/4308648/a718a01b-2378-40bb-bdf5-8faa3ee1bd2c)

**Reply by eduardogoncalves** [reply]
@rjnls  @andrem0  for some reason, if I add some role i can login with the account create using de Sign in flow.

![image](https://github.com/nextauthjs/next-auth/assets/4308648/b20c3434-b5c6-4dc2-a75a-57e27d0ac43c)
![image](https://github.com/nextauthjs/next-auth/assets/4308648/402e91bb-1f20-4805-95f1-3297d13d3fff)
![image](https://github.com/nextauthjs/next-auth/assets/4308648/79ff4509-6eed-4331-83c7-6b1723492231)

**Reply by rjnls** [reply]
correct! It's a known issue that you have to add some administrator role, even something silly like printer administrator. It's an immature product so far but I'm convinced they'll fix it.
